src/libraries/System.Text.Json/tests/JsonNode/JsonNodeOperatorTests.cs

```diff
@@ -0,0 +1,219 @@
+            Assert.NotNull((double?)(JsonValue)(double)42);
+            Assert.NotNull((decimal?)(JsonValue)(decimal)42);
+            Assert.NotNull((DateTime?)(JsonValue)new DateTime(2019, 1, 30, 12, 1, 2, DateTimeKind.Utc));
+            Assert.NotNull((DateTimeOffset?)(JsonValue)new DateTimeOffset(2019, 1, 30, 12, 1, 2, new TimeSpan(1, 0, 0)));
+            Assert.NotNull((Guid?)(JsonValue)new Guid("1B33498A-7B7D-4DDA-9C13-F6AA4AB449A6"));
+        }
+
+        [Fact]
+        public static void ImplicitOperators_FromNullValues()
+        {
+            Assert.Null((JsonValue?)(byte?)null);
+            Assert.Null((JsonValue?)(short?)null);
+            Assert.Null((JsonValue?)(int?)null);
+            Assert.Null((JsonValue?)(long?)null);
+            Assert.Null((JsonValue?)(sbyte?)null);
+            Assert.Null((JsonValue?)(ushort?)null);
+            Assert.Null((JsonValue?)(uint?)null);
+            Assert.Null((JsonValue?)(ulong?)null);
+            Assert.Null((JsonValue?)(char?)null);
+            Assert.Null((JsonValue?)(string)null);
+            Assert.Null((JsonValue?)(bool?)null);
+            Assert.Null((JsonValue?)(float?)null);
+            Assert.Null((JsonValue?)(double?)null);
+            Assert.Null((JsonValue?)(decimal?)null);
+            Assert.Null((JsonValue?)(DateTime?)null);
+            Assert.Null((JsonValue?)(DateTimeOffset?)null);
+            Assert.Null((JsonValue?)(Guid?)null);
+        }
+
+        [Fact]
+        public static void ImplicitOperators_FromNullableValues()
+        {
+            Assert.NotNull((JsonValue?)(byte?)42);
+            Assert.NotNull((JsonValue?)(short?)42);
+            Assert.NotNull((JsonValue?)(int?)42);
+            Assert.NotNull((JsonValue?)(long?)42);
+            Assert.NotNull((JsonValue?)(sbyte?)42);
+            Assert.NotNull((JsonValue?)(ushort?)42);
+            Assert.NotNull((JsonValue?)(uint?)42);
+            Assert.NotNull((JsonValue?)(ulong?)42);
+            Assert.NotNull((JsonValue?)(char?)'a');
+            Assert.NotNull((JsonValue?)(bool?)true);
+            Assert.NotNull((JsonValue?)(float?)42);
+            Assert.NotNull((JsonValue?)(double?)42);
+            Assert.NotNull((JsonValue?)(decimal?)42);
+            Assert.NotNull((JsonValue?)(DateTime?)new DateTime(2019, 1, 30, 12, 1, 2, DateTimeKind.Utc));
+            Assert.NotNull((JsonValue?)(DateTimeOffset?)new DateTimeOffset(2019, 1, 30, 12, 1, 2, new TimeSpan(1, 0, 0)));
+            Assert.NotNull((JsonValue?)(Guid?)new Guid("1B33498A-7B7D-4DDA-9C13-F6AA4AB449A6"));
+        }
+
+        [Fact]
+        public static void CastsNotSupported()
+        {
+            // Since generics and boxing do not support casts, we get InvalidCastExceptions here.
+            Assert.Throws<InvalidOperationException>(() => (byte)(JsonNode)(long)3); // narrowing
+            Assert.Throws<InvalidOperationException>(() => (long)(JsonNode)(byte)3); // widening
```

Comment:
What do the exception messages say?

Reply:
"A value of type 'System.Int64' cannot be converted to a 'System.Byte'." for the first
"A value of type 'System.Byte' cannot be converted to a 'System.Int64'." for the second

I'll add comment\Assert.